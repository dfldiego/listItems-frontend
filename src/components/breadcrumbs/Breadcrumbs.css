.breadcrumbs_ul {
  padding: 0;
  margin: 0;
}
.breadcrumbs_li {
  display: inline-block;
}
.breadcrumbs_li:not(:last-of-type)::after {
  content: "<";
  font-size: 1.4rem;
  margin: 0 5px;
  color: var(--grey);
}
.breadcrumbs_item {
  font-size: 1.4rem;
  text-decoration: none;
  color: var(--grey);
}
.breadcrumbs_item_bold {
  font-size: 1.4rem;
  text-decoration: none;
  color: var(--grey-moderate);
}
