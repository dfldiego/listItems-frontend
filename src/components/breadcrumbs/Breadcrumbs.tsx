import "./Breadcrumbs.css";
import useItems from "../../hooks/useItems";

const Breadcrumbs = () => {
  const {
    itemsState: { breadcrumbs },
  } = useItems();

  const renderBreadcrumbs = (index: number, item: string) => {
    if (index === breadcrumbs.length - 1) {
      return <span className="breadcrumbs_item_bold">{item}</span>;
    }
    return <span className="breadcrumbs_item">{item}</span>;
  };

  return (
    <>
      <ul
        className="breadcrumbs_ul"
        data-testid="breadcrumbs-component-test-id"
      >
        {breadcrumbs.length > 0 &&
          breadcrumbs.map((item: any, index: number) => (
            <li key={item} className="breadcrumbs_li">
              {renderBreadcrumbs(index, item)}
            </li>
          ))}
      </ul>
    </>
  );
};

export default Breadcrumbs;
