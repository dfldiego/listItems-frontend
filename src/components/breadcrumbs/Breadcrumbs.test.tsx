import { render, screen } from "@testing-library/react";
import Breadcrumbs from "./Breadcrumbs";

// useItems (mock de custom hook con useContext)
jest.mock("../../hooks/useItems.tsx", () => {
  return () => ({
    itemsState: {
      breadcrumbs: ["comida", "postre"],
    },
  });
});

describe("<Breadcrumbs>", () => {
  test("should render correctly", () => {
    const { container } = render(<Breadcrumbs />);
    expect(container).toMatchSnapshot();
  });

  test("should render an item", () => {
    render(<Breadcrumbs />);
    expect(screen.getByText(/comida/i)).toBeInTheDocument();
    expect(screen.getByText(/postre/i)).toBeInTheDocument();
  });
});
