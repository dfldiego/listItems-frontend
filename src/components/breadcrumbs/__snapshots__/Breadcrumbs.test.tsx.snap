// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs> should render correctly 1`] = `
<div>
  <ul
    class="breadcrumbs_ul"
    data-testid="breadcrumbs-component-test-id"
  >
    <li
      class="breadcrumbs_li"
    >
      <span
        class="breadcrumbs_item"
      >
        comida
      </span>
    </li>
    <li
      class="breadcrumbs_li"
    >
      <span
        class="breadcrumbs_item_bold"
      >
        postre
      </span>
    </li>
  </ul>
</div>
`;
